consensus/currency.go

```diff
@@ -94,6 +95,12 @@ func (c Currency) RoundDown(n uint64) (y Currency) {
 	return
 }
 
+// ContractTax returns the amount of Currency that would be taxed from a
+// contract payout with value c.
+func (c Currency) ContractTax() Currency {
+	return c.MulFloat(SiafundPortion).RoundDown(SiafundCount)
+}
```

Comment:
I would think it could it could just add the outputs after it makes the file contract, that's how transactions work. You need to fill out the signature, call CoveredFields, and then sign the transaction